src/EnergyPlus/AirflowNetworkBalanceManager.cc

```diff
@@ -1120,14 +1120,59 @@ namespace AirflowNetworkBalanceManager {
                     success = false;
                 }
 
                 ++i;
             }
         }
 
+        // *** Read AirflowNetwork simulation specified flow components
+        CurrentModuleObject = "AirflowNetwork:MultiZone:SpecifiedFlowRate";
+        state.dataAirflowNetworkBalanceManager->AirflowNetworkNumOfSFR =
+            state.dataInputProcessing->inputProcessor->getNumObjectsFound(state, CurrentModuleObject); // Temporary workaround
+        instances = state.dataInputProcessing->inputProcessor->epJSON.find(CurrentModuleObject);
+        if (instances != state.dataInputProcessing->inputProcessor->epJSON.end()) {
+            int i_mass = 0; // Temporary workaround that increasingly looks like the long term solution
+            int i_vol = 0;
+            auto &instancesValue = instances.value();
+
+            instancesValue = instances.value();
+            for (auto instance = instancesValue.begin(); instance != instancesValue.end(); ++instance) {
+                auto const &fields = instance.value();
+                auto const &thisObjectName = UtilityRoutines::MakeUPPERCase(instance.key());
+                state.dataInputProcessing->inputProcessor->markObjectAsUsed(CurrentModuleObject, instance.key()); // Temporary workaround
+
+                Real64 flow_rate{fields.at("air_flow_value")};
+                bool is_mass_flow = true;
+                if (fields.find("air_flow_units") != fields.end()) {
+                    if (fields.at("air_flow_units") != "kg/s") {
+                        is_mass_flow = false;
+                    }
+                }
+
+                // Check for name overlaps
+                if (solver.elements.find(thisObjectName) != solver.elements.end()) {
+                    ShowSevereError(state, std::string{RoutineName} + "Duplicated airflow element names are found = " + thisObjectName);
+                    success = false;
+                }
+
+                if (is_mass_flow) {
+                    state.dataAirflowNetwork->SpecifiedMassFlowData.emplace_back();
+                    state.dataAirflowNetwork->SpecifiedMassFlowData[i_mass].name = thisObjectName;
+                    state.dataAirflowNetwork->SpecifiedMassFlowData[i_mass].mass_flow = flow_rate;
+                    solver.elements[thisObjectName] = &state.dataAirflowNetwork->SpecifiedMassFlowData[i_mass]; // Yet another workaround
+                    ++i_mass;
+                } else {
+                    state.dataAirflowNetwork->SpecifiedVolumeFlowData.emplace_back();
+                    state.dataAirflowNetwork->SpecifiedVolumeFlowData[i_vol].name = thisObjectName;
+                    state.dataAirflowNetwork->SpecifiedVolumeFlowData[i_vol].volume_flow = flow_rate;
+                    solver.elements[thisObjectName] = &state.dataAirflowNetwork->SpecifiedVolumeFlowData[i_vol]; // Yet another workaround
+                    ++i_vol;
+                }
+            }
+        }
+
```

Comment:
Read the input object into the appropriate object. This would be simpler and look a lot nicer if real constructors were the order of things, but alas, no.